File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/inscriptionref/ConfirmationTokenCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.inscriptionref;

import java.util.List;
import ma.zs.alc.bean.core.inscriptionref.ConfirmationToken;
import ma.zs.alc.dao.criteria.core.inscriptionref.ConfirmationTokenCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ConfirmationTokenCollaboratorService extends  IService<ConfirmationToken,ConfirmationTokenCriteria>  {

    List<ConfirmationToken> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);




}

File: frontend-bqckup/src/app/shared/service/collaborator/inscriptionref/ConfirmationTokenCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {ConfirmationTokenDto} from '../../../model/inscriptionref/ConfirmationToken.model';
import {ConfirmationTokenCriteria} from '../../../criteria/inscriptionref/ConfirmationTokenCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class ConfirmationTokenCollaboratorService extends AbstractService<ConfirmationTokenDto, ConfirmationTokenCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/confirmationToken/';
    }

    public constrcutDto(): ConfirmationTokenDto {
        return new ConfirmationTokenDto();
    }

    public constrcutCriteria(): ConfirmationTokenCriteria {
        return new ConfirmationTokenCriteria();
    }
}
